tests/unit/test_app_endpoints.py

```diff
@@ -0,0 +1,121 @@
+"""Unit tests for app endpoints."""
+
+import os
+from unittest.mock import patch
+
+import pytest
+from fastapi import HTTPException
+
+from ols import constants
+from ols.app.endpoints import feedback, ols
+from ols.app.models.models import FeedbackRequest, LLMRequest
+from ols.utils import config
+
+
+@pytest.fixture(scope="module")
+@patch.dict(os.environ, {"LOG_LEVEL": "INFO"})
+@patch.dict(os.environ, {"BAM_API_KEY": "BAM API KEY"})
+def load_config():
+    """Load config before unit tests."""
+    config.load_config_from_env()
+
+
+def test_feedback_request(load_config):
+    """Test feedback request."""
+    feedback_request = FeedbackRequest(
+        conversation_id=123,
+        feedback_object='{"rating": 5, "comment": "Great service!"}',
+    )
+    response = feedback.feedback_request(feedback_request)
+    assert response == {"status": "feedback received"}
+
+
+@patch("ols.src.query_helpers.question_validator.QuestionValidator.validate_question")
+@patch("ols.src.docs.docs_summarizer.DocsSummarizer.summarize")
+@patch("ols.src.query_helpers.yaml_generator.YamlGenerator.generate_yaml")
+@patch("ols.utils.config.conversation_cache.get")
+def test_ols_request(
+    mock_conversation_cache_get,
+    mock_generate_yaml,
+    mock_summarize,
+    mock_validate_question,
+    load_config,
+):
+    """Test ols request."""
+    # valid question, no yaml
+    mock_validate_question.return_value = [constants.VALID, constants.NOYAML]
+    mock_summarize.return_value = (
+        "Kubernetes is an open-source container-orchestration system...",  # summary
+        "",  # referenced documents
+    )
+    llm_request = LLMRequest(query="Tell me about Kubernetes")
+    response = ols.ols_request(llm_request)
+    assert (
+        response.response
+        == "Kubernetes is an open-source container-orchestration system..."
+    )
+
+    # valid question, yaml
+    mock_validate_question.return_value = [constants.VALID, constants.YAML]
+    mock_generate_yaml.return_value = "content: generated yaml"
+    llm_request = LLMRequest(query="Generate a yaml")
+    response = ols.ols_request(llm_request)
+    assert response.response == "content: generated yaml"
+
+    # valid question, yaml, generator failure
+    mock_validate_question.return_value = [constants.VALID, constants.YAML]
+    mock_generate_yaml.return_value = constants.SOME_FAILURE
+    with pytest.raises(HTTPException) as excinfo:
+        llm_request = LLMRequest(query="Generate a yaml")
+        response = ols.ols_request(llm_request)
+        assert excinfo.status_code == 500
```

Comment:
there are named constants in requests.codes

Reply:
hmm does it try to perform real HTTP request?

Reply:
I will update the PR using requests.codes.
It does not perform real HTTP request, just run the handler function's instructions with some mocks.

Reply:
nice